File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/LogOutResponse.java
package com.skedgo.tripkit.booking;

import com.google.gson.annotations.JsonAdapter;

import org.immutables.value.Value;

import static org.immutables.gson.Gson.TypeAdapters;
import static org.immutables.value.Value.Immutable;
import static org.immutables.value.Value.Style;

@TypeAdapters
@Immutable
@Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersLogOutResponse.class)
public abstract class LogOutResponse {
    /**
     * Indicates whether user data was removed or not.
     */
    @Value.Default
    public boolean changed() {
        return false;
    }
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/LogOutResponse.kt
package com.skedgo.tripkit.booking

import com.google.gson.annotations.JsonAdapter
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Default
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@TypeAdapters
@Immutable
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersLogOutResponse::class
)
abstract class LogOutResponse {
    /**
     * Indicates whether user data was removed or not.
     */
    @Default
    open fun changed(): Boolean {
        return false
    }
}